scripts/download_image.py

```diff
@@ -123,87 +151,162 @@ def download_image(url, path, method='GET', validate=False,
     return FutureResult(url, path)
 
 
-def main(argv):
+def validate_image_url(url: str, path: str,
+                       _timeout: int = 1000) -> FutureResult:
     """
-    Download image file in URL list.
+    Validates any image url
+    :param _timeout: Connection timeout between client and server
+    :param url: URL to be check
+    :param path: not used
+    :return: url and path back for reporting
     """
-    parser = _create_parser()
-    options = parser.parse_args(argv)
-    script_dir = os.path.abspath(os.path.dirname(__file__))
-    txt_source = os.path.join(script_dir, "../txt_source")
+    try:
+        url_request = urllib.request.Request(url=url, method='HEAD')
+        urllib.request.urlopen(url_request, timeout=_timeout)
+    except HTTPError as error:
+        logging.error(
+            "Bad Url %s at path %s with error %s" % (url, path, error))
+    return FutureResult(url, path)
 
-    if not os.path.isdir(txt_source):
-        logging.error("no directory found: %s", txt_source)
-        return
 
-    key = "%s-%s" % (options.target, options.site)
-    separator = {"wiki": "=", "game": "/"}[options.site]
-    filename = os.path.join(txt_source, TXT_SOURCE[key])
-    report = REPORT_TYPE.get(options.reporter, _progress_reporter)
+def validate_source_files(source_location: str, filename: str) -> bool:
+    if not os.path.isdir(source_location):
+        logging.error("No directory found: %s", source_location)
+        return False
 
     if not os.path.isfile(filename):
         logging.error("No url list file found: %s", filename)
-        return
+        return False
+
+    return True
+
 
-    if not options.output:
-        options.output = os.path.join(script_dir, SAVE_DIR[options.target])
-        logging.info("Set default save directory: %s", options.output)
+def main(argv: list):
+    """
+    Entry point of the download image script
+    :param argv: Console arguments
+    :return:
+    """
+    script_dir = os.path.abspath(os.path.dirname(__file__))
+    source_location = os.path.join(script_dir, "../txt_source")
 
-    if not os.path.isdir(options.output):
-        logging.info("Save directory is created: %s", options.output)
-        makedirs(options.output)
+    parser = _create_parser()
+    options = parser.parse_args(argv)
+
+    key = "%s-%s" % (options.target, options.site)
+    filename = os.path.join(source_location, TXT_SOURCE[key])
+
+    if not validate_source_files(source_location,
+                                 filename) or not _create_save_locations(
+        options.output, options.target, script_dir):
+        return
 
     def transform_wiki_url(file_name):
         url = r'http://gbf-wiki.com/attach2/%s_%s.png'
         return url % ('img'.encode('utf-8').hex().upper(),
                       file_name.encode('utf-8').hex().upper())
 
-    def scan_file_for_download_list(url_map):
+    def scan_file_for_download_list(url_map: map, site: str, output: str,
```

Comment:
map is builtin function in python, not type.
map type is data type in another language, and it's dictionary for python.
typing.Iterable but list is also ok, since it not checked at run time.

Reply:
Changed to list